### PR TITLE
Sb 24484 questionare

### PR DESCRIPTION
**Pull request by rejneesh1**

### CODE REVIEW

File: src/app/client/src/app/modules/observation/components/observation-details/observation-details.component.ts [hunk 1]
```diff
@@ -124,15 +124,15 @@ export class ObservationDetailsComponent implements OnInit {
     });
   }
 
-  redirectToQuestions() {
-    //TODO: add params
+  redirectToQuestions(submissionNumber,evidenceCode) {    
+    console.log(this.selectedEntity)
```
Comment (kiranharidas187) [inline src/app/client/src/app/modules/observation/components/observation-details/observation-details.component.ts:128]: rm

File: src/app/client/src/app/modules/questionnaire/components/input-type-date-picker/input-type-date-picker.component.ts [hunk 2]
```diff
@@ -22,5 +22,9 @@ export class InputTypeDatePickerComponent implements OnInit {
   onChange(e) {
     let value = e.target.value;
     this.question.value = value;
+    this.question.startTime = this.question.startTime
```
Comment (kiranharidas187) [inline src/app/client/src/app/modules/questionnaire/components/input-type-date-picker/input-type-date-picker.component.ts:25]: do startTime in ngonInit

File: src/app/client/src/app/modules/questionnaire/components/input-type-number/input-type-number.component.ts [hunk 3]
```diff
@@ -25,6 +25,10 @@ export class InputTypeNumberComponent implements OnInit {
   onChange(e) {
     let value = e.target.value;
     this.question.value = value;
+    this.question.startTime = this.question.startTime
```
Comment (kiranharidas187) [inline src/app/client/src/app/modules/questionnaire/components/input-type-number/input-type-number.component.ts:28]: onInot do this check

File: src/app/client/src/app/modules/questionnaire/components/input-type-checkbox/input-type-checkbox.component.ts [hunk 4]
```diff
@@ -31,6 +31,10 @@ export class InputTypeCheckboxComponent implements OnInit {
     }
     this.question.value =
       this.questionnaireForm.controls[this.question._id].value;
+    this.question.startTime = this.question.startTime
```
Comment (kiranharidas187) [inline src/app/client/src/app/modules/questionnaire/components/input-type-checkbox/input-type-checkbox.component.ts:34]: startTime should be initilized on init for all components

File: src/app/client/src/app/modules/questionnaire/questionnaire.service.ts [hunk 5]
```diff
@@ -35,10 +42,10 @@ export class QuestionnaireService {
     let sections = evidence.sections;
     let answers = this.getSectionData(sections, formValues);
     let payloadData = {
-      externalid: "OB", //todo
+      externalId: evidence.externalId,
       answers: answers,
-      // startTime: 1620977187792, //todo
-      // endTime: 1620977187792, //todo
+      startTime: evidence.startTime,
+      endTime: Date.now(),
```
Comment (kiranharidas187) [inline src/app/client/src/app/modules/questionnaire/questionnaire.service.ts:48]: should we do this here?  Cant we do in init of respective components?

File: src/app/client/src/app/modules/questionnaire/questionnaire/questionnaire.component.ts [hunk 6]
```diff
@@ -17,19 +24,46 @@ export class QuestionnaireComponent implements OnInit {
...
     this.initConfiguration();
+    this.activatedRoute.queryParams.subscribe((params) => {
+      this.queryParams = params;
+      // this.configurationMode = _.get(this.queryParams, "type");
+      this.getQuestionnare();
+    });
     this.questionnaireForm = this.fb.group({});
-    this.evidence = this.data.result.assessment.evidences[0];
-    this.sections = this.evidence.sections;
+    // this.evidence = this.data.result.assessment.evidences[0];
```
Comment (kiranharidas187) [inline src/app/client/src/app/modules/questionnaire/questionnaire/questionnaire.component.ts:48]: rm unwanted codes

File: src/app/client/src/app/modules/questionnaire/questionnaire.service.ts (hunk 5 above)
Comment (rejneesh1) [reply src/app/client/src/app/modules/questionnaire/questionnaire.service.ts:48]: this. is evidence start time (which we set when we call the get details api) and end time (which we set before making make api) so i am doing in service.

**Pull request merged**